core/src/main/java/org/elasticsearch/index/query/PercolateQueryBuilder.java

```diff
@@ -459,4 +472,83 @@ public Weight createNormalizedWeight(Query query, boolean needsScores) throws IO
         }
     }
 
+    private static PercolateQuery.QueryStore createStore(PercolatorFieldMapper.PercolatorFieldType fieldType,
+                                                         QueryShardContext context,
+                                                         boolean mapUnmappedFieldsAsString) {
+        return ctx -> {
+            LeafReader leafReader = ctx.reader();
+            BinaryDocValues binaryDocValues = leafReader.getBinaryDocValues(fieldType.getQueryBuilderFieldName());
```

Comment:
In theory it is possible, so I'll add a null check.

But I think that in practise this will not happen since we will only verify docs that have query terms indexed fields and those docs also have a query builder dv field